src/cairo_run.rs

```diff
@@ -37,9 +38,16 @@ pub fn cairo_run<'a>(
 		Box::new(bigint!(execution_uudi.as_u128())),
 	);
 
-	cairo_runner
-		.run_until_pc(end, &mut vm, hint_processor)
-		.map_err(CairoRunError::VirtualMachine)?;
+	let execution_result = cairo_runner.run_until_pc(end, &mut vm, hint_processor);
+
+	match cairo_runner.exec_scopes.get_any_boxed_ref(EXPECT_REVERT_FLAG) {
+		Ok(_) if execution_result.is_ok() => Err(CairoRunError::VirtualMachine(
+			VirtualMachineError::CustomHint(EXPECT_REVERT_FLAG.to_string()),
+		))?,
+		Err(_) if execution_result.is_err() =>
+			execution_result.map_err(CairoRunError::VirtualMachine)?,
+		_ => {},
```

Comment:
in the case `Ok(_) if execution_result.is_err()` we do `{}`, which mean we don't handle the actual execution error coming from `cairo_runner.run_until_pc(end, &mut vm, hint_processor);`

I would do
```
	let res_execution_result = cairo_runner.run_until_pc(end, &mut vm, hint_processor);
	let revert_flag = cairo_runner.exec_scopes.get_any_boxed_ref(EXPECT_REVERT_FLAG).is_ok();
	
	match res_execution_result {
	  Ok() if revert_flag => Err(VirtualMachineError::CustomHint(EXPECT_REVERT_FLAG.to_string())),
	_ => {res_execution_result}
	}.map_err(CairoRunError::VirtualMachine)?;